### PR TITLE
Add umami analytics

### PR DESCRIPTION
**Pull request by SYangster**
- using umami analytics with self hosted script
- add events to track activity for different sections

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: web/public/scripts/ua_latest.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      b = "true" === f(m + "exclude-search"),
+      v = f(m + "domains") || "",
+      w = v.split(",").map(t => t.trim()),
+      S = `${(p || "https://api-gateway.umami.dev" || l.src.split("/").slice(0, -1).join("/")).replace(/\/$/, "")}/api/send`,
+      N = `${e}x${a}`,
+      T = /data-umami-event-([\w-_]+)/,
+      A = m + "umami-event",
+      x = 300;
+
+    const O = t => {
```
Comment (YuanTingHsieh) [inline web/public/scripts/ua_latest.js:32]: do we have documentation on what these methods do?

Comment (SYangster) [reply web/public/scripts/ua_latest.js:32]: no this is their tracking script, but self-hosted to avoid content blockers

Comment (YuanTingHsieh) [reply web/public/scripts/ua_latest.js:32]: do you get this script from somewhere? can we attach the link to this PR for reference, thanks!

Comment (SYangster) [reply web/public/scripts/ua_latest.js:32]: https://cloud.umami.is/script.js

**Review by YuanTingHsieh — APPROVED** [review]
Thanks LGTM

**Pull request merged**